.github/workflows/build-lint-test.yml

```diff
@@ -0,0 +1,55 @@
+# This workflow will do a clean install of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+name: Node.js CI - Build + Lint + Test
+
+on: pull_request
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Use Node.js v16
+              uses: actions/setup-node@v2
+              with:
+                  node-version: '16.x'
+                  cache: 'npm'
+
+            - name: 📦 Install dependencies
+              run: npm ci
+
+            - name: 🎨 Run ESLint on JS/TS files
+              run: npm run lint
+
+            - name: 🏗️ Build the lib
+              run: npm run build
+
+            - name: 🧪 test the build
+              run: npm run test
+
+    test-suites:
+        strategy:
+            matrix:
+                package:
+                    ['helpers', 'IPFS', 'LSP2', 'LSP3', 'LSP4', 'LSP5', 'LSP6', 'LSP12', 'LSP23']
+
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Use Node.js v16
+              uses: actions/setup-node@v2
+              with:
+                  node-version: '16.x'
+                  cache: 'npm'
+
+            - name: Install dependencies
```

Comment:
This is going to re-install the dependencies and re-build for each single package.

I would suggest to cache the build in the previous step and restore it in this step, so that it is faster and consume less resources.

See these lines as examples on how to do this:

https://github.com/lukso-network/lsp-smart-contracts/blob/def913ddabf007daad8cb486c9badb10a975af75/.github/workflows/build-lint-test.yml#L34-L42

https://github.com/lukso-network/lsp-smart-contracts/blob/def913ddabf007daad8cb486c9badb10a975af75/.github/workflows/build-lint-test.yml#L93-L102